crates/tabby/src/services/chat.rs

```diff
@@ -33,9 +34,50 @@ pub struct Message {
 
 #[derive(Serialize, Deserialize, ToSchema, Clone, Debug)]
 pub struct ChatCompletionChunk {
+    id: Uuid,
+    created: u64,
+    object: &'static str,
+    model: &'static str,
+    choices: [ChatCompletionChoice; 1],
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, ToSchema)]
+pub struct ChatCompletionChoice {
+    index: usize,
+    message: ChatCompletionMessageChoice,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, ToSchema)]
+pub struct ChatCompletionMessageChoice {
+    role: String,
+    finish_reason: Option<String>,
+    delta: ChatCompletionDelta,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, ToSchema)]
+pub struct ChatCompletionDelta {
     content: String,
 }
 
+impl ChatCompletionChunk {
+    fn new(content: String, id: Uuid, created: u64, last_chunk: bool) -> Self {
+        ChatCompletionChunk {
+            id,
+            created,
+            object: "chat.completion.chunk",
+            model: "TabbyML",
+            choices: [ChatCompletionChoice {
+                index: 0,
+                message: ChatCompletionMessageChoice {
+                    role: "assistant".into(),
+                    finish_reason: last_chunk.then_some("stop".into()),
+                    delta: ChatCompletionDelta { content },
```

Comment:
The format here still doesn't match example in https://platform.openai.com/docs/api-reference/chat/streaming

Your implementation
```json
{
   "id": "fd88f7f6-1723-42e2-b30b-4ac8401b56f9",
   "created": 1703309949,
   "object": "chat.completion.chunk",
   "model": "TabbyML",
   "choices": [
      {
         "index": 0,
         "message": {
            "role": "assistant",
            "finish_reason": null,
            "delta": {
               "content": " you"
            }
         }
      }
   ]
}
```

Example in https://platform.openai.com/docs/api-reference/chat/streaming
```json
{
   "id": "chatcmpl-123",
   "object": "chat.completion.chunk",
   "created": 1694268190,
   "model": "gpt-3.5-turbo-0613",
   "system_fingerprint": "fp_44709d6fcb",
   "choices": [
      {
         "index": 0,
         "delta": {
            "content": "Hello"
         },
         "logprobs": null,
         "finish_reason": null
      }
   ]
}
```

Reply:
Fixed, I had left out `system_fingerprint` because I wasn't sure how to generate it before. I've left it as a todo for now.

Reply:
let's use `unused-system-fingerprint` for now